### PR TITLE
pijul: 1.0.0-alpha.24 -> 1.0.0-alpha.31

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Motivation for this change

There have been several fixes to issues and the patch format was changed a bit.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/pijul/default.nix [hunk 1]
```diff
@@ -1,26 +1,28 @@
 { stdenv
+, darwin
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/pijul/default.nix:2]: Please do not use darwin as an input directly.

File: pkgs/applications/version-management/pijul/default.nix [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
 , xxHash
 , zstd
-, darwin
-, gitImportSupport ? true
-, libgit2 ? null
+# Optionals
+, gitImportSupport ? true, libgit2 ? null
 }:
 
+assert gitImportSupport -> libgit2 != null;
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/pijul/default.nix:14]:
```suggestion
```

File: pkgs/applications/version-management/pijul/default.nix [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
 , openssl
+, pkg-config
+, rustPlatform
 , xxHash
 , zstd
-, darwin
-, gitImportSupport ? true
-, libgit2 ? null
+# Optionals
+, gitImportSupport ? true, libgit2 ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/pijul/default.nix:11]:
```suggestion
, gitImportSupport ? true, libgit2
```

File: pkgs/applications/version-management/pijul/default.nix (hunk 1 above)
Comment (fabianhjr) [reply pkgs/applications/version-management/pijul/default.nix:2]: Should I import pkgs then? I am not sure what to do here.

Comment (fabianhjr) [reply pkgs/applications/version-management/pijul/default.nix:2]: The alternative from greping nixpkgs seems to be to import `pkgs` and substitute `darwin` for `pkgs.darwin`.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108506` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>pijul</li>
</ul>
</details>

File: pkgs/applications/version-management/pijul/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/version-management/pijul/default.nix:2]: That is even worse. Please do something like
```
a = callPackages ../ {
  inherit (darwin.apple_sdk.frameworks) Security;
};
```

in top-level.

Comment (fabianhjr) [reply pkgs/applications/version-management/pijul/default.nix:2]: Oh ok, yeah that seems better.

**Comment by fabianhjr** [comment]
Close due to another PR updating this package: #108767

**Pull request closed without merging**